### PR TITLE
feat: Presenter plugin.

### PR DESCRIPTION
**Pull request by richburdon**
https://github.com/dxos/dxos/assets/3523355/5997fa25-b5d5-4273-943e-49b9563a7f4f

### Summary
📝🎨🧩

This pull request adds a new package `plugin-presenter` to the `plugins` folder of the `apps` workspace. The package provides a plugin that allows the user to create and display slides using markdown syntax. The package includes the components, stories, types, and configuration files for the plugin. It also adds the ESLint, Storybook, and esbuild settings for the package.

> _`Presenter` plugin_
> _Stories and components shine_
> _Winter slides are here_

### Walkthrough
*  Add a plugin-presenter package to provide a plugin for creating and displaying slides using markdown (F0-F19)

### CODE REVIEW

File: packages/apps/plugins/plugin-ipfs/src/components/FileMain.tsx [hunk 1]
```diff
@@ -44,7 +44,22 @@ export const FileSection: FC<{ data: TypedObject }> = ({ data: file }) => {
 
   return (
     <div style={{ height }} className='p-2'>
-      <FilePreview type={file.type} url={url} />
+      <FilePreview type={file.type} url={url} className='object-contain' />
```
Comment (jessmartin) [inline packages/apps/plugins/plugin-ipfs/src/components/FileMain.tsx:47]: What options are there for file type? What kind of things can I store in IPFS?

File: packages/apps/plugins/plugin-ipfs/src/components/FilePreview.tsx [hunk 2]
```diff
@@ -4,30 +4,21 @@
...
 };
 
 /**
  * File/content preview iframe.
  */
-export const FilePreview: FC<FilePreviewProps> = ({ type, url }) => {
+export const FilePreview: FC<FilePreviewProps> = ({ type, url, className }) => {
   if (!type.startsWith('image')) {
-    return <iframe className='w-full h-full overflow-auto' src={url} />;
+    return <iframe className={mx('w-full h-full overflow-auto', className)} src={url} />;
```
Comment (jessmartin) [inline packages/apps/plugins/plugin-ipfs/src/components/FilePreview.tsx:20]: So apparently only image types can currently be rendered from IPFS. But we could extend this area to support other types of things stored in IPFS.

File: packages/apps/plugins/plugin-presenter/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# @braneframe/plugin-draw
+
+Surface plugin for sketch entities using Tldraw.
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply packages/apps/plugins/plugin-presenter/README.md:3]: Added a PR that fixes this.

File: packages/apps/plugins/plugin-presenter/package.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@braneframe/plugin-presenter",
+  "version": "0.3.1",
+  "description": "Braneframe sketch plugin",
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply packages/apps/plugins/plugin-presenter/package.json:4]: Added a PR that fixes this.Added a PR that fixes this.

**Review by jessmartin — COMMENTED** [review]
Tried out on Labs dev with some of my existing stacks. The fact that it _just works_ with existing stacks that have Markdown + Sketch is 🔥.

Nice work!

File: packages/apps/plugins/plugin-presenter/src/PresenterPlugin.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                {
+                  plugin: PRESENTER_PLUGIN,
+                  action: 'toggle-presentation',
+                },
+                {
+                  plugin: SPLITVIEW_PLUGIN,
+                  action: SplitViewAction.TOGGLE_FULLSCREEN,
+                },
+              ],
+              keyBinding: 'shift+meta+p',
```
[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/apps/plugins/plugin-presenter/src/PresenterPlugin.tsx:54]: they also need to be contextual :smile:

**Pull request merged**